Apps/WebClient/src/ClientApp/src/eventbus.ts

```diff
@@ -6,6 +6,9 @@ export default EventBus;
 export class EventMessageName {
     public static readonly CreateNote = "note-create";
     public static readonly EditNote = "note-edit";
-
     public static readonly ViewEntryDetails = "view-entry-details";
+    public static readonly RegisterBeforeUnloadWaitlistListener =
+        "waitlist-register-on-before-unload-listener";
+    public static readonly UnregisterBeforeUnloadWaitlistListener =
+        "waitlist-unregister-on-before-unload-listener";
```

Comment:
Nit: I'd suggest making the string values match the property names (kebab case is fine though).